### ISSUE TITLE
Include code frequency indicators in builder tool

### ISSUE THREAD

**Issue by sebbacon** [ebmdatalab/opencodelists#33]
When authoring a new codelist, there are diminishing returns on the time the author spends on getting every last code correct.  On a large codelist, this long tail of apparently-very-unusual-but-possibly-significant codes can take a significant amount of time to QA, possibly with very little benefit: but no-one ever finds out if the work was useful.

It would be good to have an indicator of the frequency of each code for authors to access when making decisions about code inclusion (or exclusion).

From an IG point of view it's been suggested we just start with frequencies for codes in our studies, but this doesn't help with the primary use-case, which is helping authors of new codelists (or editors of existing ones) to make decisions about adding unusual codes.

From a research point of view there's a concern that code counts could be misused for very crude studies on their own.

For the purposes of codelist authoring, centiles of a log rank would probably be sufficient: we just want to know "this code is vanishingly infrequently used". This could probably be across a single year of data aggregated across TPP and EMIS.  This may address possible concerns around data stewardship etc.

**Comment by evansd** [comment]
> centiles of a log rank would probably be sufficient

Oh nice idea.

**Comment by LiamSmeeth** [comment]
Yep, sounds good to me! Thanks

**Comment by Bengoldacre** [comment]
A key (non-IG) risk is that, while we are sharing codelist frequencies to facilitate generation and review of codelists, other people might clumsily use the resource on its own to do bad research on prevalence or trends, e.g. without recognising that these are event code counts not patient counts.

So also fine would be to list the codes with broad ordinal categories for frequency of appearance ever, e.g. 0-100, 100-10,000, 10,000-1,000,000, 1,000,000 up.

**Comment by HelenCEBM** [comment]
If using the above NHSD counts we should note that
- the figures represent total usage
- total usage may not reflect TPP usage 
- we cannot necessarily extract all codes from opensafely-TPP as some information is held in different tables

**Comment by alexwalkerepi** [comment]
Yes good to document those to be clear, but I think it's fine even with those caveats. All that's really needed is a ballpark for whether a code gets used commonly.

Also, in theory codelists will be increasingly used outside of TPP too.

**Comment by brianmackenna** [comment]
Update : NHS Digital have continued to release annual files. @LFISHER7 has made a handy browser

https://lfisher7-snomed-time-series-app-czzgeo.streamlit.app/

**Comment by Jongmassey** [comment]
@milanwiedemann FYI
